packages/react-reconciler/src/ReactFiberNewContext.new.js

```diff
@@ -294,6 +301,114 @@ export function propagateContextChange(
   }
 }
 
+export function propagateCacheRefresh(
+  workInProgress: Fiber,
+  renderLanes: Lanes,
+): void {
+  if (!enableCache) {
+    return;
+  }
+
+  let fiber = workInProgress.child;
+  if (fiber !== null) {
+    // Set the return pointer of the child to the work-in-progress fiber.
+    fiber.return = workInProgress;
+  }
+  while (fiber !== null) {
+    let nextFiber;
+
+    // Visit this fiber.
+    const list = fiber.dependencies;
+    if (list !== null) {
+      nextFiber = fiber.child;
+
+      let dependency = list.firstContext;
+      while (dependency !== null) {
+        // Check if the context matches.
+        if (dependency.context === CacheContext) {
+          // Match! Schedule an update on this fiber.
+
+          if (fiber.tag === ClassComponent) {
+            // Schedule a force update on the work-in-progress.
+            const update = createUpdate(
+              NoTimestamp,
+              pickArbitraryLane(renderLanes),
+            );
+            update.tag = ForceUpdate;
+            // TODO: Because we don't have a work-in-progress, this will add the
+            // update to the current fiber, too, which means it will persist even if
+            // this render is thrown away. Since it's a race condition, not sure it's
+            // worth fixing.
+            enqueueUpdate(fiber, update);
+          }
```

Comment:
Why do we have to special case class components here?

Reply:
It’s copied from `propagateContextChange`.

Basically a hack to override circumvent `shouldComponentUpdate`.

Reply:
Just out of interest: why is something similar not required for memo components? Its surprising to me that these work different internally :)

Reply:
We have a different, less gross mechanism for detecting context changes in a memo component :D